### DISCUSSION TITLE
Can't get UBL to work

### DISCUSSION THREAD

**Discussion by KrazyTurk** [Jyers/Marlin#1798] [Q&A] [answered]
UBL says it can't reach some of my probing points.
Got my bed leveled manually using paper, 4 corner, center. All seems good; set my z offset, upgraded my display, everything is great. When I run a new mesh using UBL 10x10 HS
Around 64-65 points it says it can't reach the rest of the spots and this is what it spits out.
I've cleared and deleted mesh and re-did it and same result.
What am I doing wrong?
And no, the end of my bed does not suddenly drop off into the abyss

![image](https://user-images.githubusercontent.com/19983954/149915031-3eeff6ea-0039-4102-ae18-303070a77516.png)

**Comment by Sonicboom247** [comment]
this is due to the restrictions from the bed size and probe offset. mine gets about 65 as well.  Just as a note of something i learned last night... Do NOT loosen the springs all the way, I was puling what little hair i had left out because the center of my prints would always come loose.  It was not until i ran the calibration print for the center and corners that i realized my springs where COMPLETELY loose.  (which was also why my Z Probe offset was a mere 2.75.  It was so bad the calibration test did the 4 outer squares and the rest was a mess i had to stop about 12 min in.  I then tightened down the screws, manually leveled about 6 times each corner, now my Z probe offset was a 2.96! reran a mesh and reprinted the calibration.  It printed perfectly this time and even with the bed cold the next morning it didn't slide off on its own.

**Comment by BlueWings172** [comment, marked as answer]
@KrazyTurk 

Are you using pre-compiled firmware or you're compiling your own and which version ?

While as @Sonicboom247 said, the UBL cannot reach all points (I use 15x15 and UBL probes only 155 points instead of 225), there an issue with the right side of your bed. UBL should compensate for it. 

But better get bed as level as much as possible as UBL compensation cannot be perfect.

Have you tried to print anything and how did the prints turn out ?

I noticed you said you leveled the bed using paper. Jyers firmware allows you to use the probe instead of paper to level you bed. this is far more accurate. In Manual Bed Leveling menu, activate use the probe and notice the reading it is showing for each corner and adjust accordingly. (don't try to get zero, so anything below 0.01 is good.)

If you don't have dual Z kit installed, sometimes the right side of the X axis will drop a bit so what you see is normal.

Also you can use Bed Visualizer plugin for Octoprint to double check the mesh. If you get red values for right side, then it looks like your X axis is dropping and you need to adjust the right eccentric nut a bit . There is many videos about this.

**Reply by KrazyTurk** [reply]
Thank you everyone for your responses.
I chose this as the "answer" only because manual leveling with the probe ... mind blown, such a game changer if you ask me. I never knew that. I'm new to Jyers, so I'm learning. I wish there was something similar for the 3Pro. I have on that I installed CR Touch on and I love it, but now I want Jyers on that too lol
